src/yunohost/app.py

```diff
@@ -640,36 +633,13 @@ def app_upgrade(app=[], url=None, file=None, force=False, no_safety_backup=False
         # Execute the app upgrade script
         upgrade_failed = True
         try:
-            upgrade_retcode = hook_exec(
-                extracted_app_folder + "/scripts/upgrade", env=env_dict
-            )[0]
-
-            upgrade_failed = True if upgrade_retcode != 0 else False
-            if upgrade_failed:
-                error = m18n.n("app_upgrade_script_failed")
-                logger.error(
-                    m18n.n("app_upgrade_failed", app=app_instance_name, error=error)
-                )
-                failure_message_with_debug_instructions = operation_logger.error(error)
-                if Moulinette.interface.type != "api":
-                    dump_app_log_extract_for_debugging(operation_logger)
-        # Script got manually interrupted ... N.B. : KeyboardInterrupt does not inherit from Exception
-        except (KeyboardInterrupt, EOFError):
-            upgrade_retcode = -1
-            error = m18n.n("operation_interrupted")
-            logger.error(
-                m18n.n("app_upgrade_failed", app=app_instance_name, error=error)
-            )
-            failure_message_with_debug_instructions = operation_logger.error(error)
-        # Something wrong happened in Yunohost's code (most probably hook_exec)
-        except Exception:
-            import traceback
-
-            error = m18n.n("unexpected_error", error="\n" + traceback.format_exc())
-            logger.error(
-                m18n.n("app_upgrade_failed", app=app_instance_name, error=error)
+            upgrade_failed, failure_message_with_debug_instructions = hook_exec_with_script_debug_if_failure(
+                extracted_app_folder + "/scripts/upgrade",
+                env=env_dict,
+                operation_logger=operation_logger,
+                error_message_if_script_failed=m18n.n("app_upgrade_script_failed"),
+                error_message_if_failed=lambda e: m18n.n("app_upgrade_failed", app=app_instance_name, error=e)
```

Comment:
I don't understand this line, especially the e variable gave to `error` arg O.o

Reply:
Eeeeh yeah sorry it's a bit odd, but it's later used in the function hook_exec_with_debug_stuff (don't remember the exact name) ... and the error message includes a sub-error message depending on the nature of the error (app script crashed, manual Ctrl+C, or exception in yunohost code) ... though most of the time it's the app script that crashed but meh ... I just tried to reproduce the current code behavior while at the same time reducing the epic lengthy toast of boring error management code